src/main/resources/com/google/api/codegen/py/enum.snip

```diff
@@ -0,0 +1,84 @@
+@snippet generateFilename(service)
+    enums.py
+@end
+
+@snippet generateBody(service)
+@end
+
+@snippet generateMethodSampleCode(docConfig)
+@end
+
+@snippet generateModule(_, __, ___)
+    @let enums = enumConstants()
+        @if enums
+            {@licenseSection()}
+        
+            {@moduleDocstring()}
+        
+        
+            {@enums}
+
+        @end
+    @end
+@end
+
+@private licenseSection()
+    @# Copyright 2016 Google Inc. All rights reserved.
+    @#
+    @# Licensed under the Apache License, Version 2.0 (the "License");
+    @# you may not use this file except in compliance with the License.
+    @# You may obtain a copy of the License at
+    @#
+    @# http://www.apache.org/licenses/LICENSE-2.0
+    @#
+    @# Unless required by applicable law or agreed to in writing, software
+    @# distributed under the License is distributed on an "AS IS" BASIS,
+    @# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    @# See the License for the specific language governing permissions and
+    @# limitations under the License.
+@end
+
+@private moduleDocstring()
+    """Wrappers for protocol buffer enum types."""
+@end
+
+@private shouldGenerateEnum(element)
+    @if element.isEnum
+        @if element.getEnumType.isReachable
```

Comment:
Why not `and(element.isEnum, element.getEnumType.isReachable)`?  Guessing arguments would be evaluated prior to `and` so that causing a NullPointerException or something?  If so, it's better to leave a comment about the situation.